src/main/java/com/datadog/api/client/ApiClient.java

```diff
@@ -1525,32 +1535,42 @@ public <T> ApiResponse<T> invokeAPI(
     Response response = null;
 
     try {
-      response = sendRequest(method, invocationBuilder, entity);
-
-      int statusCode = response.getStatusInfo().getStatusCode();
-      Map<String, List<String>> responseHeaders = buildResponseHeaders(response);
-
-      if (response.getStatusInfo() == Status.NO_CONTENT) {
-        return new ApiResponse<T>(statusCode, responseHeaders);
-      } else if (response.getStatusInfo().getFamily() == Status.Family.SUCCESSFUL) {
-        if (returnType == null) {
+      int currentRetry = 0;
+      while (true) {
+        response = sendRequest(method, invocationBuilder, entity);
+        int statusCode = response.getStatusInfo().getStatusCode();
+        Map<String, List<String>> responseHeaders = buildResponseHeaders(response);
+        if (response.getStatusInfo() == Status.NO_CONTENT) {
           return new ApiResponse<T>(statusCode, responseHeaders);
-        } else {
-          return new ApiResponse<T>(statusCode, responseHeaders, deserialize(response, returnType));
-        }
-      } else {
-        String message = "error";
-        String respBody = null;
-        if (response.hasEntity()) {
+        } else if (response.getStatusInfo().getFamily() == Status.Family.SUCCESSFUL) {
+          if (returnType == null) {
+            return new ApiResponse<T>(statusCode, responseHeaders);
+          } else {
+            return new ApiResponse<T>(
+                statusCode, responseHeaders, deserialize(response, returnType));
+          }
+        } else if (shouldRetry(currentRetry, statusCode, retry)) {
           try {
-            respBody = String.valueOf(response.readEntity(String.class));
-            message = respBody;
-          } catch (RuntimeException e) {
-            // e.printStackTrace();
+            Thread.sleep(calculateRetryIntrval(responseHeaders, retry, currentRetry) * 1000);
+            currentRetry++;
+          } catch (InterruptedException e) {
```

Comment:
What is that for? What happens if you don't catch it?

Reply:
when calling Thread.sleep() it is mandatory to catch this exception in case another thread interrupt the sleep : https://stackoverflow.com/questions/15528140/java-thread-sleep-and-interrupted-exception.